src/Core/Services/StateMigrationService.cs

```diff
@@ -314,6 +317,90 @@ private async Task MigrateFrom2To3Async()
+            var state = await GetValueAsync<State>(Storage.LiteDb, Constants.StateKey);
+            if (state?.Accounts != null)
+            {
+                string firstUserId = null;
+
+                // move values from state to standalone values in LiteDB
+                foreach (var account in state.Accounts)
+                {
+                    var userId = account.Value?.Profile?.UserId;
+                    if (userId != null)
+                    {
+                        if (firstUserId == null)
+                        {
+                            firstUserId = userId;
+                        }
+                        var vaultTimeout = account.Value?.Settings?.VaultTimeout;
+                        await SetValueAsync(Storage.LiteDb, V4Keys.VaultTimeoutKey(userId), vaultTimeout);
+
+                        var vaultTimeoutAction = account.Value?.Settings?.VaultTimeoutAction;
+                        await SetValueAsync(Storage.LiteDb, V4Keys.VaultTimeoutActionKey(userId), vaultTimeoutAction);
+
+                        var screenCaptureAllowed = account.Value?.Settings?.ScreenCaptureAllowed;
+                        await SetValueAsync(Storage.LiteDb, V4Keys.ScreenCaptureAllowedKey(userId),
+                            screenCaptureAllowed);
+                    }
+                }
+
+                // use values from first userId to apply globals
+                if (firstUserId != null)
+                {
+                    var theme = await GetValueAsync<int?>(Storage.LiteDb, V3Keys.ThemeKey(firstUserId));
+                    await SetValueAsync(Storage.LiteDb, V4Keys.ThemeKey, theme);
+
+                    var autoDarkTheme = await GetValueAsync<int?>(Storage.LiteDb, V3Keys.AutoDarkThemeKey(firstUserId));
+                    await SetValueAsync(Storage.LiteDb, V4Keys.AutoDarkThemeKey, autoDarkTheme);
+
+                    var disableFavicon =
+                        await GetValueAsync<bool?>(Storage.LiteDb, V3Keys.DisableFaviconKey(firstUserId));
+                    await SetValueAsync(Storage.LiteDb, V4Keys.DisableFaviconKey, disableFavicon);
+                }
+            }
+
+            // Update stored version
+            await SetLastStateVersionAsync(4);
+
+            // Remove old data
+            if (state?.Accounts != null)
+            {
+                foreach (var account in state.Accounts)
+                {
+                    var userId = account.Value?.Profile?.UserId;
+                    if (userId != null)
+                    {
+                        await RemoveValueAsync(Storage.LiteDb, V3Keys.ThemeKey(userId));
+                        await RemoveValueAsync(Storage.LiteDb, V3Keys.AutoDarkThemeKey(userId));
+                        await RemoveValueAsync(Storage.LiteDb, V3Keys.DisableFaviconKey(userId));
+                    }
+                }
+            }
+
+            // Removal of old state data will happen organically as state is rebuilt in app
+        }
```

Comment:
In order to avoid arrow code, I'd write it as:
```c#
if (state?.Accounts is null)
{
    // Update stored version
    await SetLastStateVersionAsync(4);
    return;
}

string firstUserId = null;
....

// Update stored version
await SetLastStateVersionAsync(4);

// Remove old data
...
```
It also helps so that you don't have to check it while removing old data